packages/react/react/src/hooks/useLazyImage.ts

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import { useRefEffect } from './useRefEffect';
+
+interface Props {
+  src: string;
+  threshold?: number | number[];
+  root?: Document | Element | null;
+  rootMargin?: string;
+  onInView?: () => void;
+}
+
+/** @tossdocs-ignore */
+export function useLazyImage({ src, rootMargin, threshold, root, onInView }: Props) {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
```

Comment:
I think the initial value of this state should be set to `false` because image loading starts when an element appears in viewport. Then:

```typescript
const imgElement = entry.target as HTMLImageElement;
imgElement.src = src;

setIsLoading(true);
imgElement.onload = () => {
  onLoad();
  setIsLoading(false);
};
```